File: introduction-to-algorithms/exercices/C/2.1-3.c
#include <stdio.h>

/*
Insertion sort algorithm exercice
Page: 22, 2.1
Exercice: 2.1-3
By: Ben Mezger
*/

int search(int *A, size_t len, int v){
    int i = 0;
    
    while (i < (int) len){
        if (A[i] == v){
            return i; 
        }
        i++;
    }
    return -1;
}

int main(){
    int A[] = { 23, 53, 12, 56, 3, 12, 68 };
    size_t len = sizeof(A) / sizeof(A[0]);
    int value_a = 12;
    int value_b = 80;
    
    int index_a = search(A, len, value_a);
    int index_b = search(A, len, value_b);
    
    if (index_a >= 0){
        printf("A: Found value %d at index %d\n", value_a, index_a);
    }
    else {
        puts("A: No value was found.\n");
    }

    if (index_b >= 0){
        printf("B: Found value %d at index %d\n", value_b, index_b);
    }
    else {
        puts("B: No value was found.\n");
    }
    
    return 0;
}

File: introduction-to-algorithms/exercices/python/2.1/2.1-3.py
#!/usr/bin/python2

# Insertion sort algorithm exercise
# Page: 22, 2.1
# Exercice: 2.1-3
# By: Ben Mezger

def search(A, v):
    i = 0

    while (i < len(A)):
        if (A[i]  == v):
            return i
        i += 1
    return -1

A = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
B = [90, 23, 5, 68, 12, 100, 5213, 12]

index_a = search(A, 8)
index_b = search(B, 0000)

print "A: index", index_a
print "B: index", index_b
